File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/TemplateDB/Template/TemplateForm.cs
using Serenity.ComponentModel;
using System;

namespace CorrespondenceSystem.TemplateDB.Forms;

[FormScript("TemplateDB.Template")]
[BasedOnRow(typeof(TemplateRow), CheckNames = true)]
public class TemplateForm
{
    [Required]
    public string Title { get; set; }
    [Required]
    public string TemplateFile { get; set; }
    public bool IsDefault { get; set; }
}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/TemplateDB/TemplateForm.ts
import { StringEditor, ImageUploadEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface TemplateForm {
    Title: StringEditor;
    TemplateFile: ImageUploadEditor;
    IsDefault: BooleanEditor;
}

export class TemplateForm extends PrefixedContext {
    static readonly formKey = 'TemplateDB.Template';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TemplateForm.init)  {
            TemplateForm.init = true;

            var w0 = StringEditor;
            var w1 = ImageUploadEditor;
            var w2 = BooleanEditor;

            initFormType(TemplateForm, [
                'Title', w0,
                'TemplateFile', w1,
                'IsDefault', w2
            ]);
        }
    }
}